### ISSUE TITLE
A summary with nested sections linking to anchors within a page breaks the prev/next buttons

### ISSUE THREAD

**Issue by netbsduser** [GitbookIO/gitbook#1831]
### What is the current behavior?

A summary with nested sections linking to anchors within a page breaks the prev/next buttons. They lead to the same page beginning

Summary.md (excerpt):

```
# Summary

## Introduction

* [The Red Book](README.md)
* [Preface](Preface.md)
  * [The Oopsilon Programming Language](Preface.md#the-oopsilon-programming-language)
  * [The Red Book](Preface.md#the-red-book)
  * [On Reading the Book](Preface.md#on-reading-the-book)
  * [For Whom the Bell Tolls](Preface.md#for-whom-the-bell-tolls)

## Part I

* [Philosophy](part-i/Philosophy.md)
  * [A Brief History of Enlightenment and Programming](part-i/Philosophy.md#history)
    * [The Enlightenment Dream](part-i/Philosophy.md#the-enlightenment-dream)
    * [Structured Programing Emerges](part-i/Philosophy.md#structured-programing-emerges)
    * [The Self-Debasing of the Enlightenment](part-i/Philosophy.md#the-self-debasing-of-the-enlightenment)
    * [Structured Programming Proves Inadequate](part-i/Philosophy.md#structured-programming-proves-inadequate)
    * [Red October](part-i/Philosophy.md#red-october)
    * [Object-oriented Programming](part-i/Philosophy.md#oop)
```

### What is the expected behavior?

Suppose one is on README.MD. Then clicking the next arrow will take you to the Preface.md. But from there the next arrow will simply take you to the to anchor "The Oopsilon Programming Language" and from there back to the same place. When the current page is any other anchor, then it will return you to "The Oopsilon Programming Language." The same behaviour occurs on other pages with nested sections - e.g. clicking on 'next' anywhere in Philosophy.md will take you to anchor #history.

**Comment by pnicoll** [comment]
We had a serious show-stopping issue with nested sections breaking the search function, which we brought to GitBook's attention in January - see https://github.com/GitbookIO/gitbook/issues/1670

**Comment by AaronO** [comment]
@pnicoll I just fixed #1670 with a `3.2.3` release (it will be deployed to `gitbook.com` shortly).

**Comment by hazenet** [comment]
I am still seeing this issue/behaviour.
Build is done via legacy.gitbook.com, which seems to be running 3.2.3
https://legacy.gitbook.com/book/hazenet/vrealize-orchestrator/

If this is not going to be fixed, does anybody know of a workaround?
